src/main/scala/common/scratchpad_adapter.scala

```diff
@@ -0,0 +1,115 @@
+package sodor.common
+
+import chisel3._
+import chisel3.util._
+import chisel3.experimental._
+
+import freechips.rocketchip.rocket._
+import freechips.rocketchip.config.{Parameters, Field}
+import freechips.rocketchip.subsystem._
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.diplomaticobjectmodel.model.OMSRAM
+import freechips.rocketchip.tile._
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.util._
+
+class SodorScratchpadAdapter(implicit p: Parameters, implicit val sodorConf: SodorConfiguration) extends Module {
+  // Parameter traits
+  val coreParams = {
+    class C(implicit val p: Parameters) extends HasCoreParameters
+    new C
+  }
+  // Extract tileParams from HasNonDiplomaticTileParameters, which is the base trait of HasCoreParameters above
+  val tileParams = coreParams.tileParams
+  // Sodor config class
+
+  // Sodor constants
+  val sodorConst = {
+    class S extends MemoryOpConstants
```

Comment:
This is to avoid name conflict between constants in Rocket and Sodor (although most of them are the same). I would clean up the constants once all cores are working.

Reply:
I just checked the Rocket constants and it seems like the operation type (encoding size and signedness) in Sodor are not there, and the read/write flag is having a different length. I cannot merge them, but here I can use the Rocket read/write flag without any problem since it is truncated correctly when assigned to the Sodor port.